src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -745,3 +745,34 @@ TEST_F(ControllerScriptEngineLegacyTest, screenWillSentRawDataIfConfigured) {
     ASSERT_ALL_EXPECTED_MSG();
 }
 #endif
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUndefinedOnUnknownCharset) {
+    const auto result = evaluate("engine.convertCharset('NULL', 'Hello!')");
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArray());
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueWellKnown) {
+    const auto result = evaluate(
+            "engine.convertCharset(engine.WellKnownCharsets.Latin9, 'Hello!')");
+
+    // int repr for 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x21
+    // aka ISO-8859-15 encoding for 'Hello!'
+    const char expected[] = {72, 101, 108, 108, 111, 33};
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArray(expected, 6));
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueStringCharset) {
+    const auto result = evaluate("`${engine.convertCharset('ISO-8859-15', 'Hello!')}`");
+    // int repr for 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x21
+    // aka ISO-8859-15 encoding for 'Hello!'
+    const char expected[] = {72, 101, 108, 108, 111, 33};
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArray(expected, 6));
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUnsupportedChars) {
+    const auto result = evaluate("`${engine.convertCharset('ISO-8859-15', 'مايأ نامز')}`");
+    // int repr for 0x1A which is ISO-8859-15 for
+    // https://en.wikipedia.org/wiki/Substitute_character
+    const char expected[] = {26, 26, 26, 26, 32, 26, 26, 26, 26};
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArray(expected, 9));
```

Comment:
hardcoding the length is ugly (and error prone), lets change that. Add a utility (feel free to change the name):
```cpp
template <int N>
QByteArray intByteArray(const char(&array)[N]) {
    return QByteArray(array, N);
}
```
```suggestion
    const QByteArray expected = intByteArray({26, 26, 26, 26, 32, 26, 26, 26, 26});
    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), expected);
```

Reply:
I think it deduces that `{26, 26, 26, 26, 32, 26, 26, 26, 26}` is list-initialization of _some_ object type and then looks at the parameter, since `const char(&array)[size]` is an array literal, it can deduce that the elements within need to be `char`s and since its a list it can deduce `size` as well (since the array has not yet [decayed](https://en.cppreference.com/w/cpp/types/decay) into a pointer). How it does it internally? don't know either, black magic. Type deduction just works 95% of the time ;) (See the beginning of Scott Mayers "Effective Modern C++" for a good explanation).

Reply:
I didn't know that C++ could do type conversion that powerful :thinking: